vpr/src/route/check_route.cpp

```diff
@@ -55,9 +55,7 @@ void check_route(enum e_route_type route_type, e_check_route_option check_route_
     const auto& rr_graph = device_ctx.rr_graph;
     auto& cluster_ctx = g_vpr_ctx.clustering();
     auto& route_ctx = g_vpr_ctx.routing();
-
-    const int num_switches = device_ctx.rr_switch_inf.size();
-
+    const int num_switches = device_ctx.rr_graph_builder.num_rr_switch();
```

Comment:
Consider to use ``RRGraphView`` here and move ``num_rr_switch()`` to ``RRGraphView``.